Source/WebCore/platform/graphics/PathStream.cpp

```diff
@@ -102,34 +96,34 @@ void PathStream::moveTo(const FloatPoint& point)
 
 void PathStream::addLineTo(const FloatPoint& point)
 {
-    auto lineTo = PathLineTo { point };
-    if (mergeIntoComposite<PathMoveTo>(lineTo))
-        return;
-    m_segments.append(lineTo);
+    if (const auto* moveTo = lastIfMoveTo())
+        m_segments.last() = { PathDataLine { moveTo->point, point } };
+    else
+        m_segments.append(PathLineTo { point });
```

Comment:
Nit - you _might_ be able to avoid some code duplication here and in the methods below, by introducing a templated helper function.